src/hooks/fetchAllMarkets.ts

```diff
@@ -0,0 +1,201 @@
+};
+
+const fetchPositionsAndBids = async (
+  isDevnet: boolean,
+  connection: Connection,
+  honeyMarketId: string,
+  honeyMarket: HoneyMarket,
+  honeyReserves: HoneyReserve[],
+  program: Program,
+) => {
+  console.log('fetching bids...');
+  const resBids = await fetch(
+    isDevnet
+      ? `https://honey-nft-api.herokuapp.com/bids/${honeyMarketId}`
+      : `https://honey-mainnet-api.herokuapp.com/bids/${honeyMarketId}`,
+    // 'http://localhost:3001/bids',
+    { mode: 'cors' },
+  );
+  const arrBids = await resBids.json();
+  const highestBid = Math.max.apply(
+    Math,
+    arrBids.map(function (o) {
+      return o.bidLimit;
+    }),
+  );
+
+  console.log('fetching positions...');
+  let arrPositions: NftPosition[] = [];
+  const solPriceUsd = await getOraclePrice(
+    isDevnet ? 'devnet' : 'mainnet-beta',
+    connection,
+    honeyReserves[0].data.switchboardPriceAggregator,
+  );
+  const nftPriceUsd = await getOraclePrice(
+    isDevnet ? 'devnet' : 'mainnet-beta',
+    connection,
+    honeyMarket.nftSwitchboardPriceAggregator,
+  );
+  const nftPrice = nftPriceUsd / solPriceUsd;
+
+  // reserve info
+  const marketReserves = await program.account.market.fetch(honeyMarket.address);
+  const reserveInfoData = new Uint8Array(marketReserves.reserves as any as number[]);
+  const reserveInfoList = MarketReserveInfoList.decode(reserveInfoData) as HoneyMarketReserveInfo[];
+
+  let obligations = await honeyMarket.fetchObligations();
+  if (obligations && reserveInfoList) {
+    await Promise.all(
+      obligations.map(async (item) => {
+        let nftMints: PublicKey[] = item.account.collateralNftMint;
+
+        const parsePosition = (position: any) => {
+          const pos: ObligationPositionStruct = {
+            account: new PublicKey(position.account),
+            amount: new BN(position.amount),
+            side: position.side,
+            reserveIndex: position.reserveIndex,
+            _reserved: [],
+          };
+          return pos;
+        };
+
+        item.account.loans = PositionInfoList.decode(Buffer.from(item.account.loans as any as number[])).map(
+          parsePosition,
+        );
+        await Promise.all(
+          nftMints.map(async (nft) => {
+            if (nft.toString() != '11111111111111111111111111111111') {
+              const totalDebt =
+                reserveInfoList[0].loanNoteExchangeRate
+                  .mul(item.account?.loans[0]?.amount)
+                  .div(new BN(10 ** 15))
+                  .div(new BN(10 ** 6)) //!!
+                  .div(new BN(10 ** 5))
+                  .toNumber() /
+                10 ** 4; //dividing lamport
+              let position: NftPosition = {
+                obligation: item.publicKey.toString(),
+                debt: totalDebt,
+                nft_mint: new PublicKey(nft),
+                owner: item.account.owner,
+                ltv: 40,
+                is_healthy: getHealthStatus(totalDebt, nftPrice),
```

Comment:
The isHealthy value is not being used anymore @bbroeking and @BowTiedHeron in this way. This still returns the old structure of RISKY, LOW, etc.

Reply:
is_healthy is something else @bowtiedpyro it checks if the NFT should be liquidated or not.
not related to the frontend.